dump.go

```diff
@@ -106,6 +106,17 @@ func (s *dumpState) dumpType(v reflect.Value) {
 }
 
 func (s *dumpState) dumpSlice(v reflect.Value) {
+	elmType := v.Type().Elem()
+	switch elmType.Kind() {
+	case reflect.Bool, reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Uintptr, reflect.Float32, reflect.Float64, reflect.String:
+		// Where the elements in a slice/array are primitive types; dump in a compact manner
+		original := s.config.Compact
+		s.config.Compact = true
```

Comment:
We should never mutate the configuration, which might be shared across multiple concurrent goroutines, and might be a constant at the caller's end. Let's add the flag to `dumpState` directly.